src/App.tsx

```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
   'Honey',
   'Jam',
   'Garlic',
-];
-
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+].map(good => ({
+  name: good,
+  id: uuidv4(),
+}));
+
+const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [value, setValue] = useState('1');
+
+  const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setValue(e.target.value);
+  };
+
+  const sortedByAlphbet = () => {
+    const sorted = [...goods].sort((a, b) => a.name.localeCompare(b.name));
+
+    setGoods(sorted);
+  };
+
+  const sortedByLength = () => {
+    const sorted = [...goods].sort((a, b) => a.name.length - b.name.length);
+
+    setGoods(sorted);
+  };
+
+  const showList = () => {
+    setIsListVisible(!isListVisible);
```

Comment:
Just `true`, if this function work like toggler it's should have name listVissibleToggler
```suggestion
    setIsListVisible(true);
```